pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/timeseries/TimeSeriesExchangeReceiveOperator.java

```diff
@@ -116,13 +118,27 @@ private TimeSeriesBlock getNextBlockWithAggregation()
     TimeBuckets timeBuckets = null;
     Map<Long, BaseTimeSeriesBuilder> seriesBuilderMap = new HashMap<>();
     Map<String, String> aggregatedStats = new HashMap<>();
+    QueryException timeoutException = null;
+
     for (int index = 0; index < _numServersQueried; index++) {
       // Step-1: Poll, and ensure we received a TimeSeriesBlock.
       long remainingTimeMs = _deadlineMs - System.currentTimeMillis();
-      Preconditions.checkState(remainingTimeMs > 0,
-          "Timed out before polling all servers. Successfully Polled: %s of %s", index, _numServersQueried);
+      if (remainingTimeMs <= 0) {
```

Comment:
I think it might be neater to handle negative `remainingTimeMs` in the `poll` method and simply return `null` there